collect_app/src/main/java/org/odk/collect/android/utilities/WidgetAppearanceUtils.java

```diff
@@ -0,0 +1,155 @@
+    public static final String COLUMNS          = "columns";
+    public static final String COLUMNS_N        = "columns-";
+    public static final String COLUMNS_FLEX     = "columns-flex";
+    public static final String QUICKCOMPACT     = "quickcompact";
+    public static final String SEARCH           = "search";
+    public static final String AUTOCOMPLETE     = "autocomplete";
+    public static final String LIST_NO_LABEL    = "list-nolabel";
+    public static final String LIST             = "list";
+    public static final String LABEL            = "label";
+    public static final String IMAGE_MAP        = "image-map";
+    public static final String NO_BUTTONS       = "no-buttons";
+    public static final String QUICK            = "quick";
+
+    // Media appearances
+    public static final String SIGNATURE        = "signature";
+    public static final String ANNOTATE         = "annotate";
+    public static final String DRAW             = "draw";
+    public static final String SELFIE           = "selfie";
+    public static final String NEW_FRONT        = "new-front";
+    public static final String NEW              = "new";
+    public static final String FRONT            = "front";
+
+    // Maps appearances
+    public static final String PLACEMENT_MAP    = "placement-map";
+    public static final String MAPS             = "maps";
+
+    // Other appearances
+    public static final String NO_APPEARANCE    = "";
+    public static final String BEARING          = "bearing";
+    public static final String EX               = "ex:";
+    public static final String THOUSANDS_SEP    = "thousands-sep";
+    public static final String PRINTER          = "printer";
+    public static final String NUMBERS          = "numbers";
+    public static final String URL              = "url";
+    public static final String RATING           = "rating";
+
+    private WidgetAppearanceUtils() {
+    }
+
+    // Get appearance hint and clean it up so it is lower case, without the search function and never null.
+    @NonNull
+    public static String getAppearance(FormEntryPrompt fep) {
+        String appearance = fep.getAppearanceHint();
+        if (appearance == null) {
+            appearance = NO_APPEARANCE;
+        } else {
+            // For now, all appearance tags are in English.
+            appearance = appearance.toLowerCase(Locale.ENGLISH);
+
+            // Strip out the search() appearance/function which is handled in ExternalDataUtil so that
+            // it is not considered when matching other appearances. For example, a file named list.csv
+            // used as a parameter to search() should not be interpreted as a list appearance.
+            appearance = ExternalDataUtil.SEARCH_FUNCTION_REGEX.matcher(appearance).replaceAll("");
+        }
+
+        return appearance;
+    }
+
+    public static int getNumberOfColumns(FormEntryPrompt formEntryPrompt, Context context) {
```

Comment:
A comment would be helpful. Something like "Gets the number of columns that choices should be displayed in. In the case of the columns-n appearance (compact-n for backwards compatibility), that number is determined from the n in the appearance string. For compact-flex, this is determined by the device screen size.

Reply:
Fixed.